src/pages/components/text-input/usage.mdx

```diff
@@ -132,99 +152,365 @@ reflect the length of the content you expect the user to enter.
+| _Warning_   | When you need to call the user's attention to an exception condition. The condition might not be an error but can cause problems if not resolved.                                                                                                                 |
+| _Disabled_  | When the user is not allowed to interact with the text input due to either permissions, dependencies, or pre-requisites. The disabled state completely removes the interactive function from a component. The styling is not subject to WCAG contrast compliance. |
+| _Skeleton_  | Use on an initial page load to indicate that the text input has not yet fully loaded.                                                                                                                                                                             |
+| _Read-only_ | _Coming soon!_                                                                                                                                                                                                                                                    |
+
+<Row>
+<Column colLg={12}>
+
+![Text input states](images/text-input-states.png)
+
+</Column>
+</Row>
+
+### Validation
+
+Real-time validation helps streamline the process and keep data clean when the
+user fills out forms. Otherwise, validate the text input data when the user
+submits the associated form. For complete guidance on error and validation, see
+the form pattern.
+
+#### Invalid
+
+An error state is triggered if the data is invalid or a required field is left
+empty. Error states have three visual indicators to signify invalid content: a
+red border, an error icon indicator, and an error message.
+
+<Row>
+<Column colLg={8}>
+
+![Example of an error state being triggered](images/text-input-usage-2.gif)
+
+</Column>
+</Row>
+
 ### Default values
 
 Where possible, add programmatic assistance. Detect and pre-fill inputs to
-reduce errors and save time. When the software can't determine the value that
+reduce errors and save time. When the software can’t determine the value that
 belongs in an input, use type-ahead to make suggestions. Use sentence-case for
 default values, detected values, and auto-completion text.
 
-### Validation and errors
+### Required vs. optional
+
+Text inputs can be labeled as either optional or required depending on the
+depending on the circumstance. For more guidance on requirement labeling, see
+the form pattern.
+
+<DoDontRow>
+<DoDont
+colMd={4} colLg={4}
+caption="If the majority of the fields are required, mark only the optional field labels with (optional).">
+
+![Example showing optional field indicator](./images/text-input-usage-optional-do.png)
```

Comment:
- These do/don't images need a side border because they bleed into the page.
- Also it seems like the arrow in the right image is not green.

<img width="794" alt="Screen Shot 2022-08-26 at 3 07 47 PM" src="https://user-images.githubusercontent.com/43969356/186974690-7daf8a7e-d4bb-48fd-af4e-84b264ba5046.png">
.